pkg/oss/utils.go

```diff
@@ -467,6 +467,85 @@ func parseDirectAssigned(runtimeClassValue, directAssignedValue string) bool {
 	return result
 }
 
+// RuntimeType represents the container runtime type
+type RuntimeType string
+
+const (
+	RuntimeTypeCOCO    RuntimeType = "coco"
+	RuntimeTypeRunD    RuntimeType = "rund"
+	RuntimeTypeRunC    RuntimeType = "runc"
+	RuntimeTypeECI     RuntimeType = "eci"
+	RuntimeTypeUnknown RuntimeType = "unknown"
+)
+
+// DetermineRuntimeType determines the container runtime type based on directAssigned, socketPath, and skipAttach.
+//
+// Support matrix:
+//   - directAssigned=true, socketPath=true, skipAttach=true: RunD
+//   - directAssigned=true, socketPath=true, skipAttach=false: error (should not occur, controller ensures empty socketPath for COCO)
+//   - directAssigned=true, socketPath=false, skipAttach=true: ECI (directAssigned not effective)
+//   - directAssigned=true, socketPath=false, skipAttach=false: COCO
+//   - directAssigned=false, socketPath=true, skipAttach=true: RunD (pure rund cluster, no need to specify directAssigned)
+//   - directAssigned=false, socketPath=true, skipAttach=false: RunC
+//   - directAssigned=false, socketPath=false, skipAttach=true: ECI
+//   - directAssigned=false, socketPath=false, skipAttach=false: error (should not occur)
+//
+// Returns:
+//   - RuntimeType: the determined runtime type
+//   - error: if the combination is invalid (should not occur scenarios)
+func DetermineRuntimeType(directAssigned bool, socketPath string, skipAttach bool) (RuntimeType, error) {
```

Comment:
I don't like this change. We don't actually care about the name of each runtime. We should have an abstract level to extract what is really different between these runtimes, and write code based on those differences. We can add comments about typical runtime, but don't depends on the runtime name in code.